core/modules/qdisp/QueryRequest.cc

```diff
@@ -76,15 +76,16 @@ QueryRequest::~QueryRequest() {
 // content of request data
 char* QueryRequest::GetRequest(int& requestLength) {
     std::lock_guard<std::mutex> lock(_finishStatusMutex);
-    if (_finishStatus != ACTIVE) {
+    auto jq = _jobQuery;
+    if (_finishStatus != ACTIVE || jq == nullptr) {
         LOGS(_log, LOG_LVL_DEBUG, _jobIdStr << " QueryRequest::GetRequest called after job finished (cancelled?)");
         requestLength = 0;
         return const_cast<char*>("");
     }
-    requestLength = _jobQuery->getDescription().payload().size();
+    requestLength = jq->getDescription().payload().size();
     LOGS(_log, LOG_LVL_DEBUG, _jobIdStr << " Requesting, payload size: " << requestLength);
     // Andy promises that his code won't corrupt it.
-    return const_cast<char*>(_jobQuery->getDescription().payload().data());
+    return const_cast<char*>(jq->getDescription().payload().data());
```

Comment:
QueryRequest::GetRequest() is an overloaded function from xrootd. We have to conform to the API.